contracts/bonding/BondingManager.sol

```diff
@@ -28,33 +29,52 @@ contract BondingManager is IBondingManager, Controllable {
-        uint256 pendingPricePerSegment; // Pending value for pricePerSegment to be set at the beginning of a new round
-        bool active;                    // Is this transcoder active. Also will be false if uninitialized
+        address transcoderAddress;                           // The address of this transcoder
+        uint256 bondedAmount;                                // The amount they have bonded themselves
+        uint256 withdrawRound;                               // The round at which delegators to this transcoder can withdraw if this transcoder resigns
+        uint256 lastRewardRound;                             // Last round that the transcoder called reward
+        uint8[3] rates;                                      // Current rates for blockRewardCut, feeShare, and pricePerSegment
+        uint8[3] pendingRates;                               // Pending rates for blockRewardCut, feeShare, and pricePerSegment
+        mapping (uint256 => TokenPools) tokenPoolsPerRound;  // Mapping of round => token pools for the round
+        bool initialized;                                    // Is this transcoder initialized
+    }
+
+    // The various states a transcoder can be in
+    enum TranscoderStatus { NotRegistered, Registered, Unbonding, Unbonded }
+
+    // Represents rewards and fees to be distributed to delegators
+    struct TokenPools {
+        RewardPool rewardPool;
+        ClaimFees[] feePool;
+    }
+
+    // Represents rewards to be distributed to delegators
+    struct RewardPool {
+        uint256 rewards;
+        uint256 transcoderTotalStake;
+    }
+
+    // Represents fees to be distributed to delegators
+    struct ClaimFees {
+        uint256 claimBlock;
+        uint256 fees;
+        uint256 transcoderTotalStake;
     }
 
     // Represents a delegator's current state
     struct Delegator {
         address delegatorAddress;          // The ethereum address of this delegator
         uint256 bondedAmount;              // The amount they have bonded
         address transcoderAddress;         // The ethereum address of the transcoder they are delgating towards
-        uint256 roundStart;                // The round the delegator transitions to bonded phase
+        uint256 startRound;                // The round the delegator transitions to bonded phase
+        uint256 delegateBlock;             // The block the delegator bonds to a transcoder
         uint256 withdrawRound;             // The round at which a delegator can withdraw
-        uint256 lastStateTransitionRound;  // The last round the delegator transitioned states
+        uint256 lastStakeUpdateRound;      // The last round the delegator transitioned states
         bool initialized;                  // Is this delegator initialized
     }
 
     // The various states a delegator can be in
-    enum DelegatorStatus { Inactive, Pending, Bonded, Unbonding }
+    enum DelegatorStatus { Inactive, Pending, Bonded, Unbonding, Unbonded }
```

Comment:
Hm I don't think there is a difference at the moment, so `Inactive` can probably just be scrapped and delegators would always default to the `Unbonded` state